src/main/java/org/testng/reporters/EmailableReporter.java

```diff
@@ -73,8 +85,12 @@ public void generateReport(List<XmlSuite> xml, List<ISuite> suites, String outdi
 
   protected PrintWriter createWriter(String outdir) throws IOException {
     new File(outdir).mkdirs();
-    return new PrintWriter(new BufferedWriter(new FileWriter(new File(outdir,
-        "emailable-report.html"))));
+    String jvmArg = System.getProperty(JVM_ARG);
+    if ((jvmArg != null) && (!jvmArg.trim().isEmpty())) {
+      fileName = jvmArg;
+    }
+
```

Comment:
Yes, its possible if one is using the TestNG API wherein they explicitly instantiate the reporter and then set the file name to null before wiring in the reporting listener via the `addListener()` method in `TestNG` class. But I don't think its possible to set this value to null using the `TestNG.main()` method. But anyways, I will add that edit check so that no possible `NullPointerException` (or) `FileNotFoundException` gets introduced. Will update the PR with that change a couple of hours later (Currently am in office. Cannot push code from here)